dht/token.cc

```diff
@@ -18,6 +18,14 @@ namespace dht {
 
 using uint128_t = unsigned __int128;
 
+inline int64_t long_token(const token& t) {
+    if (t.is_minimum() || t.is_maximum()) {
+        return std::numeric_limits<int64_t>::min();
```

Comment:
I was going to suggest we could rewrite 't.is_minimum() || t.is_maximum()' as t._kind != kind::key, but compilers are smart enough to do that using bitwise.

```
f(token&):                            # @f(token&)
        test    dword ptr [rdi], -3
        jne     .LBB0_2
        movabs  rax, -9223372036854775808
        ret
.LBB0_2:
        mov     rax, qword ptr [rdi + 8]
        ret
```

-3 means only bit 1 is zeroed, which refers to key kind.